src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.cs

```diff
@@ -39,47 +101,87 @@ out ImmutableArray<(BoundExpression, BoundDagTemp)> bindings)
             return rootIdentifier;
         }
 
-        /// <summary>
-        /// Used to create a decision dag for a switch statement.
-        /// </summary>
-        /// <param name="syntax"></param>
-        /// <param name="switchExpression"></param>
-        /// <param name="switchSections"></param>
-        /// <param name="defaultLabel"></param>
-        /// <returns></returns>
-        public BoundDecisionDag CreateDecisionDag(
+        private BoundDecisionDag CreateDecisionDag(
             SyntaxNode syntax,
-            BoundExpression switchExpression,
-            ImmutableArray<BoundPatternSwitchSection> switchSections,
-            LabelSymbol defaultLabel)
+            BoundExpression switchGoverningExpression,
+            ImmutableArray<BoundPatternSwitchSection> switchSections)
         {
-            ImmutableArray<PartialCaseDecision> cases = MakeCases(switchExpression, switchSections);
-            BoundDecisionDag dag = MakeDecisionDag(syntax, cases, defaultLabel);
+            ImmutableArray<PartialCaseDecision> cases = MakeCases(switchGoverningExpression, switchSections);
+            BoundDecisionDag dag = MakeDecisionDag(syntax, cases);
             return dag;
         }
 
-        private ImmutableArray<PartialCaseDecision> MakeCases(BoundExpression switchExpression, ImmutableArray<BoundPatternSwitchSection> switchSections)
+        private ImmutableArray<PartialCaseDecision> MakeCases(BoundExpression switchGoverningExpression, ImmutableArray<BoundPatternSwitchSection> switchSections)
         {
-            var rootIdentifier = new BoundDagTemp(switchExpression.Syntax, switchExpression.Type, null, 0);
+            var rootIdentifier = new BoundDagTemp(switchGoverningExpression.Syntax, switchGoverningExpression.Type, null, 0);
             int i = 0;
             var builder = ArrayBuilder<PartialCaseDecision>.GetInstance(switchSections.Length);
             foreach (BoundPatternSwitchSection section in switchSections)
             {
                 foreach (BoundPatternSwitchLabel label in section.SwitchLabels)
                 {
-                    builder.Add(MakePartialCaseDecision(++i, rootIdentifier, label));
+                    if (label.Syntax.Kind() != SyntaxKind.DefaultSwitchLabel)
+                    {
+                        builder.Add(MakePartialCaseDecision(++i, rootIdentifier, label));
```

Comment:
>++i [](start = 60, length = 3)

Why 1-based rather than 0-based? Please add a comment to `PartialCaseDecision.Index`.

Reply:
The index is not used in any way except in the hash code of `PartialCaseDecision`. Its purpose is to be distinct from one case to another to distinguish cases. It is also monotonically increasing as a way of prioritizing the patterns (earlier match first), but it turns out that feature wasn't needed. We could just as well use Syntax.Location.Start in the hash code and eliminate this field, but it is useful for debugging too.

Comment will be added in next iteration.

---
In reply to: [172603783](https://github.com/dotnet/roslyn/pull/24059#discussion_r172603783) [](ancestors = 172603783)